improve Mesh Service Installation Guide (#117)
Volue-Public/energy-smp-docs: docs/mesh/installation/MeshServiceInstallationGuide.md

```json
  "SharedPointCache": {
    "LimitMb": 1000, // -1 means no limit, valid range is [0, 1048576]
    "DefaultIntervalDays": 20, // -1 means no limit, valid range is [1, 20000]
    "PreloadAllTimeSeries": false,
    "PreloadPreviousState": false,
    "CacheStateFileDirectory": "<path>"
  },
```

> Note: this is just an example. Specific values must be fine-tuned depending on

Reviewer: ```suggestion
**_Note!_** This is just an example. Specific values must be fine-tuned depending on
```

```json
  "SharedPointCache": {

>     "LimitMb": 1000, // -1 means no limit, valid range is [0, 1048576]

Reviewer 2: The error messages and logged configuration reports `LimitMb = -1` as `"LimitMb": null`. And similarly the reader converts `"LimitMb": null` to the internal `LimitMb = -1`. We should probably stick to using `null` instead of `-1` in docs.

Author: I'm fine either way. Thanks for checking!

For security reasons, access to `CacheStateFileDirectory` should be restricted at the operating-system level.
Grant permissions only to the specific users or groups that require access (principle of least privilege).

> #### Monitoring and tuning the cache

Reviewer: ```suggestion
#### Monitor and tune the cache
```

Reviewer: Do we need to update this in a TOC as well?

Author: ToC updated. Thanks for suggestion!

- `evictedPoints` is the number of time series points that have been evicted
  from the cache.
- `inserts` is the number of times a time series has been inserted into the
  cache.
- `updates` is the number of updates to individual time series that have been
  performed. This only counts updates that hit the cache, updates to time
  series not in the cache are not counted.
- `stalls` is the number of times cache retrieval has stalled because of high
  cache pressure.
- `pointLimit` is the approximate maximum number of points in the cache based
  on the `LimitMb` configuration.
- `expansions` is the number of times Mesh had to fetch additional points for a
  time series that already had some points cached. Useful for tuning `DefaultIntervalDays`.
- `estimatedMemoryKb` is approximately how many KB the cache is using.

When monitoring these values and tuning cache performance you should watch for
a few problematic scenarios.

1. A continuous "rapid" increase in `evictedSeries` and `misses` may indicate
   that the cache is too small for the existing workload. The ratio of increase
   of hits to increase of misses is a good indication of how bad the issue is.

> 2. `stalls` should not happen in normal usage. A two digit increase in stalls
>    every now and then is fine, but if `stalls` are happening often they should
>    be reported to the development team as they may indicate a current or future
>    performance issue.

Reviewer: ```suggestion
2. `stalls` should not happen in normal usage. A two digit increase in stalls
   every now and then is fine, but if `stalls` are happening often, they should
   be reported to the development team as they may indicate a current or future
   performance issue.
```

Reviewer: Maybe "reported to Volue support" as this document also targets customers?

Author: Good point! Applied.

#### How the cache works

The cache is implemented as a least recently used (LRU) cache. This means that

> when the cache is over-full the time series that are evicted are those that

Reviewer: ```suggestion
when the cache is over-full, the time series that are evicted are those that
```

were least recently accessed. While this pattern is simple to implement and
understand it is one that struggles when the cache is too small. For example

> if the cache has space for 100 entires but there's a repeating workload

Reviewer: ```suggestion
if the cache has space for 100 entries but there is a repeating workload
```

> accessing 101 entries in a sequence the cache will always evict the next

Reviewer: ```suggestion
accessing 101 entries in a sequence, the cache will always evict the next
```

item to be accessed.

> This is why it's important to monitor `evictedSeries` and `misses` and to

Reviewer: ```suggestion
This is why it is important to monitor `evictedSeries` and `misses` and to
```

For security reasons, access to `CacheStateFileDirectory` should be restricted at the operating-system level.
Grant permissions only to the specific users or groups that require access (principle of least privilege).

#### Monitor and tune the cache

When `LimitMb` is set, you may monitor the state of the cache via the following

> new fields in the Mesh health endpoint, in the group `metrics/pointCache/`.

Reviewer 2: ```suggestion
fields in the Mesh health endpoint, in the group `metrics/pointCache/`.
```
